lib/api/impl/healthcare_element_api_impl.dart

```diff
@@ -97,4 +97,56 @@ class HealthcareElementApiImpl extends HealthcareElementApi {
   Future<List<String>?> matchHealthcareElement(Filter filter) {
     return api.baseHealthElementApi.rawMatchHealthElementsBy(filter.toAbstractFilterDto());
   }
+
+  @override
+  Future<HealthcareElement> giveAccessTo(HealthcareElement healthcareElement, String delegatedTo) async {
+    final localCrypto = api.crypto;
+    final currentUser = await api.baseUserApi.getCurrentUser();
+
+    // Check if delegatedBy has access
+    if (!healthcareElement.systemMetaData!.delegations.entries.any((element) => element.key == currentUser!.dataOwnerId())) {
+      throw StateError("Couldn't give access to unowned healthcareElement");
```

Comment:
StateError("DataOwner $currentUser.dataOwner() does not have the right to access healthcare element $healthcareElement.id")